Tests/SonarScanner.MSBuild.PreProcessor.Test/SonarQubeWebServiceTest.cs

```diff
@@ -0,0 +1,423 @@
+﻿/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Google.Protobuf;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarScanner.MSBuild.Common;
+using SonarScanner.MSBuild.PreProcessor.Protobuf;
+using SonarScanner.MSBuild.PreProcessor.Test.Infrastructure;
+using SonarScanner.MSBuild.PreProcessor.WebService;
+using TestUtilities;
+
+namespace SonarScanner.MSBuild.PreProcessor.Test
+{
+    [TestClass]
+    public class SonarQubeWebServiceTest
```

Comment:
This is the previous `SonarWebServiceTest` class that we renamed to `SonarQubeWebServiceTest` and from here we moved some of the tests to the base and sonar cloud web service tests.